### PR TITLE
tests/hostname: Limit to fcos only

### PR DESCRIPTION
**Pull request by cgwalters**
It never affected RHEL.

### CODE REVIEW

File: tests/kola/networking/hostname/fallback-hostname/test.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -xeuo pipefail
 
-# kola: { "platforms": "qemu" }
+# kola: { "distros": "fcos", "platforms": "qemu" }
```
Comment (travier) [inline tests/kola/networking/hostname/fallback-hostname/test.sh:4]: Can you update the list below to mention the new fcos only option? Thanks

Comment (cgwalters) [reply tests/kola/networking/hostname/fallback-hostname/test.sh:4]: Done; though I definitely disagree with the duplication inherent in this comment format

Comment (cgwalters) [reply tests/kola/networking/hostname/fallback-hostname/test.sh:4]: OK yeah I disagree so much with the copy-pasta inherent in this that this made me do https://github.com/coreos/coreos-assembler/pull/3059

**Comment by dustymabe** [comment]
hmm. so yes, the change at https://fedoraproject.org/wiki/Releases/37/ChangeSet#Fallback_Hostname only landed in Fedora/FCOS, but the root problem never affected RHEL/RHCOS, so this test should be valid there too.

So the real question: was this test failing? If it was failing it was probably because of the way the test is written and could be modified to run on RHCOS too. i.e. maybe we modify the test to only check one or two things rather than 4.

**Comment by HuijingHei** [comment]
Agree with you @dustymabe , maybe can merge this to let the CI run smoothly for rhcos, and do some tweaks to make it work for RHCOS in the follow PR.

**Comment by dustymabe** [comment]
> Agree with you @dustymabe , maybe can merge this to let the CI run smoothly for rhcos, and do some tweaks to make it work for RHCOS in the follow PR.

Sounds good, but let's update the comment to be accurate (i.e. including the information from my previous comment).

**Comment by travier** [comment]
We're not failing in RHCOS CI yet for this test, only when bumping the FCOS submodule: https://github.com/openshift/os/pull/965

**Comment by travier** [comment]
And this does not fail for SCOS/RHCOS 9 apparently so maybe we need to skip it on RHCOS 8.

**Comment by travier** [comment]
Did that in https://github.com/openshift/os/pull/965 so I think we can close this one.

**Comment by cgwalters** [comment]
> And this does not fail for SCOS/RHCOS 9 apparently so maybe we need to skip it on RHCOS 8.

Yeah, that's fair.  This points to the obvious issue of our "distro" tag being too weak, what we really want is more like conditions `"osver": "rhel >= 9"` or so.  But doing that would require some work in kola.

**Pull request closed without merging**